app/containers/Forms/ProductField.tsx

```diff
@@ -25,16 +27,22 @@ interface Props extends FieldProps<FormData> {
 export const ProductFieldComponent: React.FunctionComponent<Props> = (
   props
 ) => {
-  const {formData, query, onChange} = props;
+  const {formData, query, naicsCode, onChange} = props;
 
-  // TODO: Clean up this function with array.some() as noted in https://github.com/bcgov/cas-ciip-portal/pull/621
-  const productIsPublished = (formData, query) => {
-    const product = query.allProducts.edges.find(
-      ({node}) => node.rowId === formData.productRowId
-    )?.node;
-    if (product?.productState === 'PUBLISHED' || !product) return true;
-    return false;
-  };
+  const productIsPublished = (formData, query) =>
+    query.allProducts.edges.some(
+      ({node}) =>
+        node.rowId === formData.productRowId &&
+        node.productState === 'PUBLISHED'
+    ) || !formData.productRowId;
+
+  const productInNaicsCode = (formData, naicsCode) =>
+    naicsCode?.allProductsByNaicsCode.edges.some(
+      (edge) => edge.node.rowId === formData.productRowId
+    ) || !formData.productRowId;
+
+  const hasSelectableProducts = (naicsCode) =>
+    naicsCode?.allProductsByNaicsCode?.edges.length > 0;
```

Comment:
I don't think that those need to be functions.
Variable name shadowing is also a code smell, which we need to take care of by adding the corresponding eslint rule (in a different PR): https://eslint.org/docs/rules/no-shadow